Fix: Retrieve Sandbox and Deployment IDs from CLI (EOSU-668)

JessTello opened the pull request:
- Added new class to fetch Sandbox ID and Deployment ID from CLI arguments
- Replaced previous logic that only read values from JSON after initialization
- Ensures correct IDs are used when running builds in different environments

JessTello committed:
Fix: Retrieve Sandbox and Deployment IDs from CLI (EOSU-668)

- Added new class to fetch Sandbox ID and Deployment ID from CLI arguments
- Replaced previous logic that only read values from JSON after initialization
- Ensures correct IDs are used when running builds in different environments

matt-clarke commented on com.playeveryware.eos/Runtime/Core/EOSCommandLineOverrides.cs:
```diff
@@ -0,0 +1,91 @@
+    {
+        try
+        {
+            Debug.Log($"[{nameof(EOSCommandLineOverrides)}::{nameof(ApplyCliOverrides)}] Initializing CLI overrides�");
```
Invalid char in message
```suggestion
            Debug.Log($"[{nameof(EOSCommandLineOverrides)}::{nameof(ApplyCliOverrides)}] Initializing CLI overrides");
```

JessTello replied:
Comment applied

matt-clarke commented on com.playeveryware.eos/Runtime/Core/EOSCommandLineOverrides.cs:
```diff
@@ -0,0 +1,91 @@
+        public static string DeploymentIdOverride { get; set; }
+        public static string SandboxIdOverride { get; set; }
+    }
+}
```
Namespace should apply to all types in the file

JessTello replied:
Comment applied

matt-clarke commented on com.playeveryware.eos/Runtime/Core/EOSCommandLineOverrides.cs:
```diff
@@ -0,0 +1,91 @@
+                    deploymentArg = arg.Substring(DeploymentKey.Length);
+
+                if (arg.StartsWith(SandboxKey, StringComparison.OrdinalIgnoreCase))
+                    sandboxArg = arg.Substring(SandboxKey.Length);
```
```suggestion
                if (arg.StartsWith(DeploymentKey, StringComparison.OrdinalIgnoreCase))
                    deploymentArg = arg.Substring(DeploymentKey.Length);
                else if (arg.StartsWith(SandboxKey, StringComparison.OrdinalIgnoreCase))
                    sandboxArg = arg.Substring(SandboxKey.Length);
```

JessTello replied:
Comment applied

matt-clarke commented on com.playeveryware.eos/Runtime/Core/EOSCommandLineOverrides.cs:
```diff
@@ -0,0 +1,91 @@
+            }
+
+            // If no overrides found, exit early
+            if (deploymentArg == null && sandboxArg == null)
```
Adding `IsNullOrEmpty()` adds a little extra safety
```suggestion
            if (string.IsNullOrEmpty(deploymentArg) && string.IsNullOrEmpty(sandboxArg))
```

JessTello replied:
Comment applied

matt-clarke commented on com.playeveryware.eos/Runtime/Core/EOSCommandLineOverrides.cs:
```diff
@@ -0,0 +1,91 @@
+            // Save values globally for later access
+            // --------------------------------------------------------------------
+            EOSOverrideState.DeploymentIdOverride = deploymentArg;
+            EOSOverrideState.SandboxIdOverride = sandboxArg;
```
I'm a bit confused about the need for both these override vars and updating the platform config below. If the override vars are used then why is there a need to update the platform config? 

With that in mind I would get rid of the override vars and just update the config. This eliminates the chance that someone might read the values from the config without checking the overrides.

JessTello replied:
The main reason is that PlatformManager.GetPlatformConfig() is called in several places, and every time this function runs, it restores and reads the values from the JSON file. I would need to do a deeper analysis to understand why it was implemented this way and whether it’s truly necessary. However, to avoid any risk, I decided to take the safer approach and store the values so that EOSManager reads them directly, instead of using the ones that are constantly overwritten when PlatformManager.GetPlatformConfig() is called.

Additionally, with this approach, if the CLI is not used, the platform still falls back to the values from the platform JSON.

JessTello replied:
By the way, the reason for applying them now is to prepare for a future refactor of PlatformManager.GetPlatformConfig(). Once that method stops overwriting values, this script will already handle applying them, so the overrides could be removed without extra work. That’s why I didn’t remove the function but added an explicit comment for clarity.

matt-clarke replied:
That's very strange, `PlatformManager.GetPlatformConfig()` uses the static var `s_platformConfig`. Once it's been called once it should always use that cached version.

Regardless changing the `GetPlatformConfig()` function to apply the cmd line args would be safest, which can be done straight after `s_platformConfig` is assigned. This way whenever `GetPlatformConfig()` is called then these overrides are applied, even if something were to clear `s_platformConfig` since it was last called.

JessTello replied:
You were right, Matt. Thank you so much for pointing that out. It’s now implemented in the new commit.

matt-clarke commented on com.playeveryware.eos/Runtime/Core/EOSCommandLineOverrides.cs:
```diff
@@ -0,0 +1,91 @@
+
+            // --------------------------------------------------------------------
+            // Regenerate a clean platform config
+            // --------------------------------------------------------------------
```
This comment isn't accurate and is unnecessary, as there's already the comment below. Also, I'm not a fan of the extra comment lines for just regular comments

JessTello committed:
Refactor: Apply PR review fixes and cleanup

- Fixed invalid char in debug log
- Applied namespace globally
- Replaced nested if-if with if-else
- Updated null checks to use string.IsNullOrEmpty
- Removed inaccurate and redundant comments

JessTello committed:
Refactor: Apply command-line overrides in GetPlatformConfig

- Ensure overrides are applied before s_platformConfig assignment
